### ISSUE TITLE
false-positive 'mkdir(...)' race condition

### ISSUE THREAD

**Issue by freezy-sk** [kalessil/phpinspectionsea#994]
| Subject    | Details                                                  |
| :--------- | :------------------------------------------------------- |
| Issue type | Bug/False-Positive     |
| IDE        | PhpStorm 2018.2.1 Build #PS-182.3911.43                          |
| Plugin     | Php Inspections EA Extended 3.0.5.1 |

Description or screenshot:
--
![image](https://user-images.githubusercontent.com/661637/44023676-6191c60e-9eeb-11e8-94a4-a58b15e200f6.png)

Inspection is reporting error when it's split into separate `if` statements

It's not reporting error only when it's written in one `if`
```php
if (!is_dir($path) && !mkdir($path, $mode) && !is_dir($path))
```

Environment details
---
PhpStorm 2018.2.1
Build #PS-182.3911.43, built on August 8, 2018
Licensed to Code Less Mess s. r. o. / Marek Viger
Subscription is active until May 12, 2019
JRE: 1.8.0_152-release-1248-b8 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6

**Comment by kalessil** [comment]
Thank you for reporting @freezy-sk. I'm not fully understanding the context, would you please share some insights regarding the code fragment? So far looks like like it was intended following:

```php
if (!is_dir($path) && !mkdir($path, $mode) && !is_dir($path)) {
    throw new \Exception('...');
}
return true;
```
but I'm not getting why try-catch is there.

**Comment by freezy-sk** [comment]
hi @kalessil 
it looks like that piece of code was inspired in this function https://github.com/yiisoft/yii2/blob/master/framework/helpers/BaseFileHelper.php#L613
but regardless of that try-catch part, it should not report false-positive when it's split into separate if statements

**Comment by kalessil** [comment]
Checked, unfortunately I couldn't extract a reliable criteria for this false positive. Hence suppression is a recommended way to go ahead.

**Comment by freezy-sk** [comment]
thank you @kalessil
